internal/host/statemachine.go

```diff
@@ -312,6 +312,24 @@ func NewHostStateMachine(sm stateswitch.StateMachine, th *transitionHandler) sta
 		PostTransition:   th.PostUnbindHost,
 	})
 
+	sm.AddTransition(stateswitch.TransitionRule{
```

Comment:
What about the transition from unbinding-pending-user-action to discovering-unbound?

Reply:
My thought is that the only transition out of this state is to re-register the host.
Then the old host instance will be deleted from the DB and the state machine will start from scratch.

WDYT?

Reply:
The only time when the host is deleted from DB is when DeregisterHost is called. Upon any other re-registration , the host will exist in DB and we must make sure that there is a valid transition from its current state to a new one (discovering-unbound)

Reply:
So, the host is not deleted, need to handle the transition